vmhooks/smallIntOps.md

```diff
@@ -9,20 +9,37 @@ require "../elrond-config.md"
 module SMALLINTOPS
     imports BASEOPS
     imports ELROND-CONFIG
+    imports private LIST-BYTES-EXTENSIONS
 
     // extern long long smallIntGetUnsignedArgument(void *context, int32_t id);
-    rule <instrs> hostCall("env", "smallIntGetUnsignedArgument", [ i32 .ValTypes ] -> [ i64 .ValTypes ])
-               => #returnIfUInt64(Bytes2Int(ARGS[ARG_IDX], BE, Unsigned), "argument out of range") ... </instrs>
-         <locals> 0 |-> <i32> ARG_IDX </locals>
-         <callArgs> ARGS </callArgs>
-      requires #validArgIdx(ARG_IDX, ARGS)
+    rule [smallIntGetUnsignedArgument]:
+        <instrs> hostCall("env", "smallIntGetUnsignedArgument", [ i32 .ValTypes ] -> [ i64 .ValTypes ])
+              => #if #validArgIdx(ARG_IDX, ARGS)
```

Comment:
For symbolic execution, I would split this rule in two an put `#validArgIdx(ARG_IDX, ARGS)` in the `requires` clause. The main reason is that if this function is not fully evaluated, the entire `if` is not evaluated, and execution can't continue.